### ISSUE TITLE
Issue reproducing Base-to-Novel results on FGVCAircraft and EuroSAT (High variance on Novel classes)

### ISSUE THREAD

**Issue by DragonBoyL** [llcllc1997/FCPrompt#2]
Body:
Hi authors,

Thanks for the great work and for open-sourcing the codebase!

I am currently trying to reproduce the Base-to-Novel generalization results. While my reproduced Base accuracies align very closely with the reported numbers in the paper, I am observing a noticeable performance drop and unusually high variance specifically on the Novel classes for the FGVCAircraft and EuroSAT datasets.

Please note that I have already strictly followed the dataset-specific hyperparameters provided in your paper's table for the Base-to-novel setting. Specifically:

For FGVCAircraft, I used: epochs = 25, \rho = 0.05, K = 12.

For EuroSAT, I used: epochs = 50, \rho = 0.1, K = 6.

Even with these specific settings, the gap remains:

For FGVCAircraft, my reproduced Base accuracy is 43.92% (very close to the reported 43.68%). However, on the Novel class, my average accuracy is only 31.51% (compared to the reported 36.97%). Furthermore, the variance across my 3 runs is quite high: 38.09%, 20.64%, and 35.81%, resulting in a mean of 31.51% ± 7.74%.

I observed a similar pattern on EuroSAT. The Base accuracy matches well (96.41% vs. reported 96.80%), but the Novel accuracy drops to an average of 71.14% (compared to the reported 75.21%). Across 3 runs, the Novel accuracies were 70.38%, 67.59%, and 75.46%, yielding a mean of 71.14% ± 3.26%.

For context, I am running this on Ubuntu using a single RTX 4090. The command I used for training is as follows:

Bash
  bash ./scripts/test_new.sh fgvc_aircraft 4
Given these results, I have a few quick questions to help close this gap:

Random Seeds: Are there specific random seeds you recommend for these two datasets to stabilize the Novel class accuracy?

Undocumented Tweaks: Beyond the hyperparameters explicitly listed in the table (\rho, K, epochs), are there any other undocumented tweaks (e.g., learning rate scaling, prompt length, or text prompt templates) used specifically for FGVCAircraft and EuroSAT?

Any guidance or suggestions would be greatly appreciated. Thank you for your time!

**Comment by llcllc1997** [comment]
Thank you for your interest in our work! We are sorry for not providing the generalizable version of FCPrompt in this repository. And since the original GPU platform are not longer used, we are not able to fully provide the files for you at this time.

First, 
1) Random seeds are set to 1, 2, 3 for all datasets and experiments.
2) No undocumented tweaks for FGVCAircraft and EuroSAT.

Second, here may be some potential suggestions for reproducing the b2n experiments.
1) Batch size should be 4 for b2n and 32 for few-shot setting as we mentioned in the paper.
2) How do you implement the SAM optimizer? You can find the SAM-based optimizer in that repository if you need (https://github.com/llcllc1997/GCSCoOp).

Hope they are helpful!
